test/cbmc/proofs/UDP/vProcessGeneratedUDPPacket_IPv6/vProcessGeneratedUDPPacket_IPv6_harness.c

```diff
@@ -0,0 +1,131 @@
+/* Standard includes. */
+#include <stdint.h>
+#include <stdio.h>
+
+/* FreeRTOS includes. */
+#include "FreeRTOS.h"
+#include "task.h"
+#include "queue.h"
+#include "list.h"
+#include "semphr.h"
+
+/* FreeRTOS+TCP includes. */
+#include "FreeRTOS_IP.h"
+#include "FreeRTOS_ND.h"
+#include "FreeRTOS_UDP_IP.h"
+#include "FreeRTOS_IP_Private.h"
+
+/* Include the stubs for APIs. */
+#include "memory_assignments.c"
+#include "freertos_api.c"
+
+/* We do not need to calculate the actual checksum for the proof to be complete.
+ * Neither does the checksum matter for completeness. */
+uint16_t usGenerateChecksum( uint16_t usSum,
+                             const uint8_t * pucNextData,
+                             size_t uxByteCount )
+{
+    __CPROVER_assert( pucNextData != NULL, "Next data in GenerateChecksum cannot be NULL" );
+
+    uint16_t usChecksum;
+
+    /* Return any random value of checksum since it does not matter for CBMC checks. */
+    return usChecksum;
+}
+
+
+/* We do not need to calculate the actual checksum for the proof to be complete.
+ * Neither does the checksum matter for completeness. */
+uint16_t usGenerateProtocolChecksum( const uint8_t * const pucEthernetBuffer,
+                                     size_t uxBufferLength,
+                                     BaseType_t xOutgoingPacket )
+{
+    __CPROVER_assert( pucEthernetBuffer != NULL, "The Ethernet buffer cannot be NULL while generating Protocol Checksum" );
+    uint16_t usProtocolChecksum;
+
+    /* Return random value of checksum since it does not matter for CBMC checks. */
+    return usProtocolChecksum;
+}
+
+/* This function has been tested separately. Therefore, we assume that the implementation is correct. */
+eARPLookupResult_t eNDGetCacheEntry( IPv6_Address_t * pxIPAddress,
+                                     MACAddress_t * const pxMACAddress,
+                                     struct xNetworkEndPoint ** ppxEndPoint )
+{
+    __CPROVER_assert( pxIPAddress != NULL, "pxIPAddress cannot be NULL" );
+    __CPROVER_assert( pxMACAddress != NULL, "pxMACAddress cannot be NULL" );
+
+    eARPLookupResult_t eResult;
+
+    return eResult;
+}
+
+/* This function has been tested separately. Therefore, we assume that the implementation is correct. */
+void vNDSendNeighbourSolicitation( NetworkBufferDescriptor_t * pxNetworkBuffer,
+                                   const IPv6_Address_t * pxIPAddress )
+{
+    __CPROVER_assert( pxNetworkBuffer != NULL, "The network buffer descriptor cannot be NULL." );
```

Comment:
Same here, should we be asserting that the ethernet buffer in the network buffer is also non-NULL?